ballista/scheduler/src/flight_sql.rs

```diff
@@ -196,26 +196,48 @@ impl FlightSqlServiceImpl {
     ) -> Result<Vec<FlightEndpoint>, Status> {
         let mut fieps: Vec<_> = vec![];
         for loc in completed.partition_location.iter() {
-            let (host, port) = if let Some(ref md) = loc.executor_meta {
+            let (exec_host, exec_port) = if let Some(ref md) = loc.executor_meta {
                 (md.host.clone(), md.port)
             } else {
                 Err(Status::internal(
-                    "Invalid partition location, missing executor metadata".to_string(),
+                    "Invalid partition location, missing executor metadata and advertise_endpoint flag is undefined.".to_string(),
                 ))?
             };
+
+            let (host, port) = match self.server.advertise_endpoint {
+                Some(_) => {
+                    let advertise_endpoint_vec: Vec<&str> = self
+                        .server
+                        .advertise_endpoint
+                        .as_ref()
+                        .expect("Failed to parse advertise_endpoint value")
```

Comment:
The endpoint is already available from the pattern matching so the `expect` code is unreachable. This can be simplified:

```suggestion
            let (host, port) = match &self.server.advertise_endpoint {
                Some(endpoint) => {
                    let advertise_endpoint_vec: Vec<&str> = endpoint
```